src/components/StepIndicator.module.scss

```diff
@@ -0,0 +1,86 @@
+@use "../styles/colors" as *;
+
+$step-number-size: 2.0625rem;
+
+.container {
+  display: flex;
+  justify-content: center;
+  gap: 1rem;
+  height: 10.75rem;
+  padding: 2rem 0 6.6875rem;
+  background-image: url("../assets/images/bg-sidebar-mobile.svg");
+  background-repeat: no-repeat;
+  background-size: cover;
+}
+
+.step {
+  display: flex;
+  flex-direction: column;
+  min-height: $step-number-size;
+}
+
+.stepNumber {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-radius: 100%;
+  border: 1px solid white;
+  color: white;
+  width: $step-number-size;
+  height: $step-number-size;
+  font-weight: 700;
+  translate:
+    color 100ms ease-in,
+    background-color 100ms ease-in;
+
+  &.active {
+    color: $marine-blue;
+    background-color: $light-blue;
+    border-width: 0;
+  }
+}
+
+.stepId {
+  display: none;
+  text-transform: uppercase;
+  color: $pastel-blue;
+}
+
+.stepName {
+  display: none;
+  margin-top: 0.25rem;
+  text-transform: uppercase;
+  font-weight: 700;
+  letter-spacing: 1px;
+  color: $white;
+}
+
+@media only screen and (min-width: 1024px) {
```

Comment:
I think it is more readable if you use @media separately for each class, so that during coding it is much easier to see how the component will change for smaller devices without need of scrolling file up and down to compare it